java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightEndpoint.java

```diff
@@ -54,13 +57,22 @@ public FlightEndpoint(Ticket ticket, Location... locations) {
    * Constructs a new endpoint with an expiration time.
    *
    * @param ticket A ticket that describe the key of a data stream.
+   * @param expirationTime (optional) When this endpoint expires.
    * @param locations  The possible locations the stream can be retrieved from.
    */
   public FlightEndpoint(Ticket ticket, Instant expirationTime, Location... locations) {
```

Comment:
Should we deprecate this constructor in favor of the builder?

Reply:
Hmm, I think it's OK to keep it